Structure_Engine/Query/MaterialComposition.cs

```diff
@@ -136,6 +147,48 @@ public static MaterialComposition MaterialComposition(this ConcreteSection secti
         }
 
         /***************************************************/
+        /**** Public Methods - Interface                ****/
+        /***************************************************/
+
+        [PreviousVersion("4.3", "BH.Engine.Structure.Query.MaterialComposition(BH.oM.Structure.SectionProperties.ISectionProperty)")]
+        [Description("Returns a SectionProperty's MaterialComposition.")]
+        public static MaterialComposition IMaterialComposition(this ISectionProperty sectionProperty)
+        {
+            return sectionProperty.IsNull() ? null : MaterialComposition(sectionProperty as dynamic);
+        }
+
+        /***************************************************/
+        /**** Private methods - Default                 ****/
+        /***************************************************/
+
+        [Description("Returns a SectionProperty's homogeneous MaterialComposition.")]
+        private static MaterialComposition MaterialComposition(this ISectionProperty sectionProperty)
+        {
+            return sectionProperty.IsNull() ? null : (MaterialComposition)Physical.Create.Material(sectionProperty.Material);
+        }
+
+        /***************************************************/
+        /**** Private methods                           ****/
+        /***************************************************/
+
+        private static MaterialComposition MaterialComposition(this IMaterialFragment baseMaterial, ReinforcementDensity reinforcementDesity)
+        {
+            if(reinforcementDesity.Material == null || reinforcementDesity.Material.Density == 0 || reinforcementDesity.Density == 0)
+                return (MaterialComposition)Physical.Create.Material(baseMaterial);
+
+            //Calculate volume of reinforcement per unit area
+            double reinfVolume = reinforcementDesity.Density / reinforcementDesity.Material.Density;
```

Comment:
There should be a check here for:
1. Whether the `reinforcementDensity.Density`  or `reinforcementDensity.Material.Density` is negative (happy for this to ignored as it's up to the user, however it may be useful to flag dud data)
2. Whether the `reinforcementDensity.Density` is equal to or more than the `reinforcementDensity.Material.Density` i.e. the ratio will be > 1.0